source/graphql/reflection.d

```diff
@@ -0,0 +1,84 @@
+module graphql.reflection;
+import graphql.traits;
+import std.traits;
+import std.meta;
+import vibe.data.json;
+
+package struct SchemaReflection(Schema)
+{
+	static SchemaReflection *instance() @safe
+	{
+		_instance.initialize();
+		return &_instance;
+	}
+
+	string[TypeInfo] classes;
+	string[][TypeInfo] derivatives;
+	string[][string] bases;
+	static struct TypeWithStrippedName
+	{
+		Json typeJson;
+		string name;
+		bool canonical;
+	}
+	TypeWithStrippedName[string] jsonTypes;
+
+
+	private:
+	static SchemaReflection _instance;
```

Comment:
`SchemaReflection` here is the instantiated template type here. You get one SchemaReflection instance per Schema type. I can't see a reason to have more than one.

Reply:
> please put this closer to the instance.

OK, I just stuck it down here so it was under the `private` label.

Reply:
what is the benefit of static then?

Reply:
It just makes it so we only calculate it once, in case it's used in multiple places. The types aren't going to change, there's no need to reevaluate anything.

I don't know what the use case is for multiple copies of the same GraphQL schema, but this matches the purpose for which this is written. The type data is not a function of the GraphQL instance, but of the schema itself, which is unchanging.